lib/oxidized/model/tplink.rb

```diff
@@ -34,27 +39,39 @@ class TPLink < Oxidized::Model
     cfg
   end
 
-  cmd 'show system-info' do |cfg|
+  cmd "show system-info" do |cfg|
     comment cfg.each_line.to_a[3..-3].join
   end
 
-  cmd 'show running-config' do |cfg|
+  cmd "show running-config" do |cfg|
     lines = cfg.each_line.to_a[1..-1]
     # cut config after "end"
     lines[0..lines.index("end\n")].join
   end
 
   cfg :telnet, :ssh do
-    username /^User ?[nN]ame:/
-    password /^\r?Password:/
+    username /^(User ?[nN]ame|User):/
+    password /^\r?[pP]assword:/
   end
 
   cfg :telnet, :ssh do
-    if vars :enable
-      post_login do
-        send "enable\r"
-        cmd vars(:enable)
+    post_login do
+      if vars(:enable) == true
+        cmd "enable"
+      elsif vars(:enable)
+        cmd "enable", /^[pP]assword:/
+        send vars(:enable) + "\n\r"
       end
+      # disable paging
+      cmd "terminal length 0"
+      # enable-admin gives admin privileges for regular users
+      # Sending enable-admin with an admin user returns a message warning that
+      # the user already is admin without further consequences. So, always
+      # send the enable-admin.
+      # There is an option to set a password for enable-admin, but testing
```

Comment:
Hi, setting the enable password DOES work. It's related to AAA configuration, there's one method list for login password and one for enable password, it needs to be set to 'local' instead of nothing.
(Testing on a `T1700G-28TQRev2` - still 2.0.1, gonna check if there's an update I need to add)

Reply:
I'll try again to configure and test the password when calling enable-admin

Reply:
I'm confused. So TP-Link has Exec, Privileged Exec and Admin Exec, and you switch from Exec to Privileged Exec with "enable" and from Privileged Exec to Admin Exec with "enable-admin", and both commands can have (different?) passwords?

Reply:
When you log in as a normal user it logs in Exec mode, then you need `enable` to enter `Enter Privileged EXEC Mode` and then use `enable-admin` to `Achieve the admin privilege`.

`enable` and `enable-admin` can have equal, different or empty passwords. Factory configuration defaults to both empty, if I remember correctly.

Reply:
And @FlorianHeigl was correct, I just had a AAA config incomplete. Now I can have both `enable` and `enable-admin` passwords set.

And I don't know where to store the `enable-admin` password, there is another device with three different passwords? Or the better is to not support this configuration?

Reply:
It will result in "Error: Bad command" on older devices. Maybe we should make it configurable?

Reply:
But, even with this `Error: Bad command`, the backup will be done on those older devices that doesn't have the `enable-admin`. Just tested sending some made up `Bad command`s and the backup ends successfully.

I guess it deserves a comment in the code stating that behaviour, so people won't be puzzled by this message.

As a new information, while testing sending invalid commands, I found that using the T1500G-10PS's newest firmware, the `enable-admin` is not needed to `show running-config` and the other information gathering commands. But I really don't know if this behaviour change is a bug or a feature. Regardless this, the `enable-admin` is still needed by some other devices and firmware versions.